server/mqtt_test.go

```diff
@@ -2160,6 +2203,28 @@ func TestMQTTPublish(t *testing.T) {
 	testMQTTPublish(t, mcp, mpr, 2, false, false, "foo", 2, []byte("msg"))
 }
 
+func TestMQTTQoS2PubReject(t *testing.T) {
+	o := testMQTTDefaultOptions()
+	o.MQTT.rejectQoS2Pub = true
+	s := testMQTTRunServer(t, o)
+	defer testMQTTShutdownServer(s)
+
+	nc := natsConnect(t, s.ClientURL())
+	defer nc.Close()
+
+	mcp, mpr := testMQTTConnect(t, &mqttConnInfo{cleanSess: true}, o.MQTT.Host, o.MQTT.Port)
+	defer mcp.Close()
+	testMQTTCheckConnAck(t, mpr, mqttConnAckRCConnectionAccepted, false)
+
+	testMQTTPublish(t, mcp, mpr, 1, false, false, "foo", 1, []byte("msg"))
+
+	testMQTTPublishNoAcks(t, mcp, 2, false, false, "foo", 2, []byte("msg"))
+	_, err := mpr.readByte("failed attempt")
+	if err == nil || !strings.Contains(err.Error(), "error reading failed attempt: EOF") {
+		t.Fatalf("Expected error about QoS 2 publish rejected, got %v", err)
+	}
+}
+
```

Comment:
Should we also add a test for a QoS2 will message in connect (and make sure that the connection fails)?

Reply:
👍 ditto for dropping retained QoS2 messages; I was just trying to get this done quickly

Reply:
@kozlovic Can you please review the added tests?